codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/EntityServiceSourceGenerator.scala

```diff
@@ -66,33 +66,50 @@ object EntityServiceSourceGenerator {
       interfaceSource(service, entity, packageName, className).getBytes(Charsets.UTF_8)
     )
 
-    val handleClassName = className + "Handler"
+    val handlerClassName = className + "Handler"
     val handlerSourcePath =
-      generatedSourceDirectory.resolve(packagePath.resolve(handleClassName + ".java"))
+      generatedSourceDirectory.resolve(packagePath.resolve(handlerClassName + ".java"))
 
-    handlerSourcePath.getParent.toFile.mkdirs()
     handlerSource(service, entity, packageName, className).foreach { doc =>
+      handlerSourcePath.getParent.toFile.mkdirs()
       Files.write(
         handlerSourcePath,
         doc.getBytes(Charsets.UTF_8)
       )
     }
 
+    val providerClassName = className + "Provider"
+    val providerSourcePath =
+      generatedSourceDirectory.resolve(packagePath.resolve(providerClassName + ".java"))
+
+    providerSource(service, entity, packageName, className).foreach { src =>
+      providerSourcePath.getParent.toFile.mkdirs()
+      Files.write(
+        providerSourcePath,
+        src.getBytes(Charsets.UTF_8)
+      )
+    }
+
     if (!implSourcePath.toFile.exists()) {
       // We're going to generate an entity - let's see if we can generate its test...
       val testClassName = className + "Test"
       val testSourcePath =
         testSourceDirectory.resolve(packagePath.resolve(testClassName + ".java"))
-      val testSourceFiles = if (!testSourcePath.toFile.exists()) {
-        testSourcePath.getParent.toFile.mkdirs()
-        Files.write(
-          testSourcePath,
-          testSource(service, entity, packageName, implClassName, testClassName).getBytes(Charsets.UTF_8)
-        )
-        List(testSourcePath)
-      } else {
-        List.empty
-      }
+      val testSourceFiles =
+        if (!testSourcePath.toFile.exists()) {
+          testSource(service, entity, packageName, implClassName, testClassName) match {
+            case Some(src) =>
+              testSourcePath.getParent.toFile.mkdirs()
+              Files.write(
+                testSourcePath,
+                src.getBytes(Charsets.UTF_8)
+              )
+              List(testSourcePath)
+            case None => List.empty
```

Comment:
I had to make this optional for now. 

We have one codegen for the tests for Value Entity and EventSourced Entity. This is now in our way, because the constructor of the entity needed to changed because of the provider. 

So, for now, we only generate the test for the EventSourced one. 

In any case, that will take another shape once we integrate @franciscolopezsancho work.